src/db/access_keys.rs

```diff
@@ -0,0 +1,63 @@
+use bigdecimal::BigDecimal;
+use crate::schema;
+use crate::db::enums::{ActionEnum, StatusEnum, PermissionEnum};
+use schema::access_keys;
+
+#[derive(Insertable, Queryable)]
+pub(crate) struct AccessKey {
+    pub public_key: String,
+    pub account_id: String,
+    pub action: ActionEnum,
+    pub status: StatusEnum,
+    pub receipt_hash: String,
+    pub block_height: BigDecimal,
+    pub permission: PermissionEnum,
+}
+
+impl AccessKey {
+    pub fn from_receipt_view(
+        receipt: &near_indexer::near_primitives::views::ReceiptView,
+        block_height: u64,
+    ) -> Vec<Self> {
+        let mut access_keys: Vec<Self> = vec![];
+        if let near_indexer::near_primitives::views::ReceiptEnumView::Action { actions, .. } =
+        &receipt.receipt
+        {
+            for action in actions {
+                let access_key = match action {
+                    near_indexer::near_primitives::views::ActionView::AddKey {
+                        public_key,
+                        access_key,
+                    } => Self {
+                        public_key: public_key.to_string(),
+                        account_id: receipt.receiver_id.to_string(),
+                        action: ActionEnum::Add,
+                        status: StatusEnum::Pending,
+                        receipt_hash: receipt.receipt_id.to_string(),
+                        block_height: block_height.into(),
+                        permission: match access_key.permission {
+                            near_indexer::near_primitives::views::AccessKeyPermissionView::FullAccess => {
+                                PermissionEnum::FullAccess
+                            },
+                            near_indexer::near_primitives::views::AccessKeyPermissionView::FunctionCall { .. } => {
+                                PermissionEnum::FunctionCall
+                            },
+                        },
```

Comment:
You can implement `from AccessKeyPermissionView for PermissionEnum`, and then use `permission: access_key.permission.into()` here and in the `main.rs` (genesis handler).